rclcpp/include/rclcpp/intra_process_manager.hpp

```diff
@@ -368,6 +370,22 @@ class IntraProcessManager
     }
   }
 
+  /// Return true if the given rmw_gid_t matches any stored Publishers.
+  bool
+  matches_any_publishers(const rmw_gid_t * id)
```

Comment:
`weak_ptr::lock` is a const operation, so this function can be const.

http://en.cppreference.com/w/cpp/memory/weak_ptr/lock